### DISCUSSION TITLE
License implementation

### DISCUSSION THREAD

**Discussion by Fail-Safe** [sqm-autorate/sqm-autorate#78] [Operational Development]
@chas-iot @Lochnair  Hoping we can move the license implementation communication over here...

**Comment by chas-iot** [comment]
Excellent!

I've just pushed a fix to the lua to remove the duplicated license header. 

@Fail-Safe @Lochnair @dlakelan please replace the @name with your preferred contact details (or remove where necessary) .

Charles

**Reply by Fail-Safe** [reply]
Thanks @chas-iot! I've update my preferred contact 👍

**Reply by dlakelan** [reply]
updated

**Comment by chas-iot** [comment]
The getstats.sh and README.md (including pix) have complex histories. For those that started working on these before I became active, to whom should we attribute copyright? Are their contributions substantial enough that we should secure permission to license as GPLv3?

Charles

Background: my understanding of copyright law is that it covers a specific expression of ideas, not the idea itself. In the free and open source software world there is a tradition of more widespread attribution.

**Comment by richb-hanover** [comment]
With the decision to use MPLv2, we should update the LICENSE files in the relevant branch(es)

**Comment by chas-iot** [comment]
I've updated the `develop/licensing` branch, ready for review. It will not be a clean merge, so after review, I plan to rebase onto `develop/main` before applying

**Comment by richb-hanover** [comment]
Comments on the `develop/licensing` branch:

- README still mentions GPL
- I tweaked the initial paragraphs of the README in the `main` branch in case anyone wandered to the site before we completed the new Organization. I hadn't realized you had done so much work here. Feel free to ignore/cherry pick anything from that branch
- The **Requirements** language has always felt a bit off... Could we change to something like:
   > This sqm-autorate program is written primarily for OpenWrt 21.02. It is not our priority to run on earlier versions, since none of the developers runs that older release. If we get reports that it's working, we may change our recommendation.
- You can remove the reference to the original shell version. (Or maybe we can start a History section that outlines the evolution.)
- Update: I updated the `getstats.sh` script in the `testing/lua-threads`branch

Thanks!

**Comment by chas-iot** [comment]
@richb-hanover too much happening all at the same time

good point about the licensing, I'll add that in the develop/new-docs branch sometime in the next 48 hours - likely much sooner

DONE
